README.md

```diff
@@ -116,31 +261,238 @@ For each region return an array of the metrics from the 'metrics gathering' sect
+        "LAX",
+        "MDW"
+      ]
+    },
+    {
+      "id": "Image to image",
+      "models": [
+        "ByteDance/SDXL-Lightning",
+        "timbrooks/instruct-pix2pix"
+      ],
+      "regions": [
+        "FRA",
+        "LAX",
+        "MDW"
+      ]
+    }
+    ...
+  ]
 }
 ```
 
-## Deployment Using Vercel
 
-These serverless functions are deployed using Vercel.
+#### `GET /api/regions`
+
+This endpoint outputs the regions in JSON format.  It does not take any parameters.
+
+#### Response 
+```
+{
+  "regions": [
+    {
+      "id": "TOR",
+      "name": "Toronto",
+      "type": "transcoding"
+    },
+    {
+      "id": "HND",
+      "name": "Tokyo",
+      "type": "transcoding"
+    },
+    {
+      "id": "SYD",
+      "name": "Sydney",
+      "type": "transcoding"
+    },
+    {
+      "id": "STO",
+      "name": "Stockholm",
+      "type": "transcoding"
+    },
+  ]
+}
+```
+
+## Database
+
+The database is responsible for storing the results of test data for each job executed as well as some reference data (regions).
+
+The new schema allows for easy addition for Regions and Job Types (AI, Transcoding, etc...)
+
+The new schema has index all queries to allow fast searching on common fields:
+orchestrator
+timestamp
+model/pipeline
+region
+job type 
+
+![Leaderboard Database Entity Relation Diagram](docs/new-db-entity-relation.png)
+
+## Migrations
+
+As reference data and schema design evolves, it is necessary to deploy these changes to your backend database.  In order to avoid human error and manual tasks, database migrations are automated in this project.  This means one can update DDL and DML in the databsae with the addition of a SQL script.  In other words, you can alter the structure of the database or the data stored in the databse with these migrations.
+
+These scripts must follow the naming convention of <migration_number>_descriptive_text>.<up | down>.sql.  As an example:
+
+```
+8_create_users_table.up.sql
+8_create_users_table.down.sql
+```
+This example defines the eighth migration for the datbaase with two migrations, one to upgrade the database and one to revert it.  When the application starts its connection to the database, it will run all upgrade (or up) migrations automatically.
+
+Database migrations are found in the assets/migrations folder.  These are embeded in the golang binary built from this project for ease of access regardless of where the application is deployed.  This also allows Vercel to use these migrations.  The migrations are loaded and processed by the [golang-migrate](https://github.com/golang-migrate/migrate) project.  Please read their documentation for more details.
+
+### Upgrading the database
+
+The database upgrade runs automatically once the first API is called.  This is because the first API call instantiates a connection to the database, which first checks that the database exists and then applies any migrations necessary to the database.
+
+### Downgrading the database
```

Comment:
The downgrade scripts are used for testing the migrations among other potential cases.  IMO, they are harmless if following best practices (most importantly, backup db before doing a downgrade).  Also, downgrading is not automated so it would a manual exercise one has to evaluate before running.

Reply:
My point is mainly, why to even include it when we don't want to use it? Do you need it for any testing scenarios?

In general, less code is better :) So, I'd try to avoid adding more code that is not used anywhere.

Reply:
I agree with less code is better.  We currently use it for testing that the migration works.  It also ensures that the new release does not impart any new steps to users of the codebase.  They can just drop it in and run.  The migrations also ensure that the database is versioned to be able to know the exact state of the database by looking at one table.